src/main/java/com/ivanmostovyi/demo/service/UserService.java

```diff
@@ -0,0 +1,17 @@
+package com.ivanmostovyi.demo.service;
+
+import com.ivanmostovyi.demo.domain.User;
+import com.ivanmostovyi.demo.dto.UserDto;
+
+import java.util.List;
+import java.util.Optional;
+
+public interface UserService {
+
+    List<User> findAll();
+
+    void create(UserDto userDto);
+
+    Optional<User> findByUsername(String username);
```

Comment:
Service souldn't return optional, in such a synchronous system exception should be thrown in case of error path, that should be handled by interceptor (in scope of this PR, can be left unhandled).